### PR TITLE
remove string_view STL dependency

### PR DESCRIPTION
**Pull request by SchrodingerZhu**

### CODE REVIEW

**Comment by SchrodingerZhu** [comment]
The CI failures are due to macos-12 being obsolete. We should probably update the images

**Comment by mjp41** [comment]
I've just pushed a PR to move the CI forward to newer versions.  Hopefully will be smooth.

**Comment by mjp41** [comment]
Rebasing should fix the CI.

File: src/snmalloc/pal/pal.h [hunk 1]
```diff
@@ -159,9 +159,7 @@ namespace snmalloc
    * The following are supported as arguments:
    *
    *  - Characters (`char`), printed verbatim.
-   *  - Strings (anything convertible to `std::string_view`), typically string
```
Comment (mjp41) [inline src/snmalloc/pal/pal.h:162]: As we've lost the other types that can flow into this, I am left wondering if we actually use this append function?

Comment (mjp41) [reply src/snmalloc/pal/pal.h:162]: As I guess I'm wondering if we should inline the functionality into append?

Comment (SchrodingerZhu) [reply src/snmalloc/pal/pal.h:162]: Sorry, was busy on other stuffs. Inling should work except the two definitions will run into ambiguity.

**Review by mjp41 — APPROVED** [review]
LGTM. Thanks for stressing the earlier review.

**Pull request merged**